pkg/collector/osmlogs_collector.go

```diff
@@ -41,72 +41,61 @@ func (collector *OsmLogsCollector) Collect() error {
 		return err
 	}
 
-	// ** Collects ground truth on OSM resources **
-	var groundTruthMap = map[string][]string{
-		"allResourcesTable":               []string{"get", "all", "--all-namespaces", "-l", "app.kubernetes.io/name=openservicemesh.io", "-o", "wide"},
-		"allResourcesConfigs":             []string{"get", "all", "--all-namespaces", "-l", "app.kubernetes.io/name=openservicemesh.io", "-o", "json"},
-		"mutatingWebhookConfigurations":   []string{"get", "MutatingWebhookConfiguration", "--all-namespaces", "-l", "app.kubernetes.io/name=openservicemesh.io", "-o", "json"},
-		"validatingWebhookConfigurations": []string{"get", "ValidatingWebhookConfiguration", "--all-namespaces", "-l", "app.kubernetes.io/name=openservicemesh.io", "-o", "json"},
-	}
-
-	for fileName, kubeCmds := range groundTruthMap {
-		if err = collector.CollectKubectlOutputToCollectorFiles(rootPath, fileName, kubeCmds); err != nil {
-			log.Printf("Failed to collect %s for OSM: %+v", fileName, err)
-		}
-	}
-
 	controllerNamespaces, err := utils.GetResourceList([]string{"get", "deployments", "--all-namespaces", "-l", "app=osm-controller", "-o", "jsonpath={..metadata.namespace}"}, " ")
 	if err != nil {
 		return err
 	}
 
 	for _, meshName := range meshList {
+		meshRootPath := filepath.Join(rootPath, "mesh_"+meshName)
 		monitoredNamespaces, err := utils.GetResourceList([]string{"get", "namespaces", "--all-namespaces", "-l", "openservicemesh.io/monitored-by=" + meshName, "-o", "jsonpath={..name}"}, " ")
 		if err != nil {
 			log.Printf("Failed to get namespaces within osm mesh '%s': %+v\n", meshName, err)
 			continue
 		}
```

Comment:
Moving controller namespace collection inside the for loop and adding the label `meshName=" + meshName` prevents duplicates.

```suggestion
		for _, meshName := range meshList {
			controllerNamespaces, err := utils.GetResourceList([]string{"get", "deployments", "--all-namespaces", "-l", "app=osm-controller,meshName=" + meshName, "-o", "jsonpath={..metadata.namespace}"}, " ")
		if err != nil {
			return err
		}
	
		meshRootPath := filepath.Join(rootPath, "mesh_"+meshName)
		monitoredNamespaces, err := utils.GetResourceList([]string{"get", "namespaces", "--all-namespaces", "-l", "openservicemesh.io/monitored-by=" + meshName, "-o", "jsonpath={..name}"}, " ")
		if err != nil {
			log.Printf("Failed to get namespaces within osm mesh '%s': %+v\n", meshName, err)
			continue
		}
```

Reply:
Update: Use `...,app.kubernetes.io/instance=" + meshName` instead.

Reply:
![image](https://user-images.githubusercontent.com/13926417/118865909-51170380-b896-11eb-8434-4742e0dfcee9.png)